src/platform/FailSafeContext.cpp

```diff
@@ -186,5 +186,11 @@ CHIP_ERROR FailSafeContext::DeleteFromStorage()
     return PersistedStorage::KeyValueStoreMgr().Delete(keyAlloc.FailSafeContextKey());
 }
 
+void FailSafeContext::ForceFailSafeTimerExpiry()
+{
+    FailSafeTimerExpired();
+    DisarmFailSafe();
```

Comment:
We have a similar difficulty with Fabric removal.  Far and away the easiest thing to do for that case is to send the response to the command, but defer actual removal by dispatching to the work queue.  However, this would allow for more enqueued items that interact with the fabric to execute _after_ fabric removal, which is obviously incorrect.

Sorry to jump into this comment.  But I just thought it was worth mentioning that we have this problem occurring on multiple fronts, and it's a difficult thing to solve.

Reply:
> FailSafeTimerExpired() schedules work that also cancels the failsafe timer. 

I'm not able to find where that happens.  Could you point me to it, please?